libraries/recipe_utils.rb

```diff
@@ -119,6 +119,27 @@ def self.separate_splunk_installed?(node)
     Dir.exist?(splunk_home(node['platform_family'], node['kernel']['machine'], opposite_package_name))
   end
 
+  # Returns Boolean for whether a Splunk artifact is already installed.
+  def self.splunk_installed?(node)
+    Dir.exist?(splunk_home(node['platform_family'], node['kernel']['machine'], node['splunk']['package']['base_name']))
+  end
+
+  # Determine if we need to use the splunk start command instead of systemctl
+  # https://docs.splunk.com/Documentation/Splunk/8.1.3/Admin/RunSplunkassystemdservice#Upgrade_considerations_for_systemd
+  def self.use_splunk_start_command?(node, previous_splunk_version) # rubocop:disable Metrics/CyclomaticComplexity
+    return false unless ::File.exist?(node['splunk']['systemd_file_location'])
+    return false if previous_splunk_version.nil?
+
+    previous_version = Gem::Version.new(previous_splunk_version)
+    current_version = Gem::Version.new(node['splunk']['package']['version'])
+
+    return true if previous_version <= Gem::Version.new('7.4.0') && current_version >= Gem::Version.new('8.0.0')
```

Comment:
7.4 seems pretty unlikely since they're on 8.1 already.  Maybe a 7.3.10, but I don't see 7.4 being a realistic possibility.  Upated in https://github.com/cerner/cerner_splunk/pull/239/commits/67497f6473b8f88def74c407902b636a01b799fa.